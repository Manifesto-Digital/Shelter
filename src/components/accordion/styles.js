import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
