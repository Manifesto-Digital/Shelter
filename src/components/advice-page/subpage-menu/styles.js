import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
