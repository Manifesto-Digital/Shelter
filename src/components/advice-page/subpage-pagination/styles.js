import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
