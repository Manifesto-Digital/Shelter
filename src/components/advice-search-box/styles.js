import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
