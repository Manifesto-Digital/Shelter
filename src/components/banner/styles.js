import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
