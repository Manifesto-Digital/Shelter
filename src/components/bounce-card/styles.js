import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
