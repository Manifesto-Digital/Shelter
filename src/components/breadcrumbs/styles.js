import styled from 'styled-components';
import LinkHandler from '../link-handler';

export const BreadcrumbList = styled.ul`
  margin: 0;
  padding: 0.8em 0;
`;

export const BreadcrumbListItem = styled.li`
  display: inline-block;
  list-style: none;
  margin: 0 0.4em 0 0;
  padding: 0;
  font-size: ${({ theme }) => theme.fontsize.small};

  &:after {
    content: '>';
    margin-left: 0.4em;
  }
  &:last-child {
    &:after {
      content: '';
    }
  }
`;

export const BreadcrumbAnchor = styled(LinkHandler)`
  text-decoration: none;
`;
