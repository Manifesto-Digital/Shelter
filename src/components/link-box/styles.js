import styled from 'styled-components';
import { Link } from 'gatsby';
import { breakpoint } from '../theme/breakpoint';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin: ${props => props.theme.spacing.largePadding} 0;
  padding: ${props => props.theme.spacing.padding};
  background: ${props => props.theme.palette.white};
  box-shadow: ${props => props.theme.boxshadow.small};
  border-radius: 3px;
`;

export const ListWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;

  ${breakpoint.desktop`
    flex-direction: row;
    flex-wrap: wrap;
  `};
`;

export const StyledLink = styled(Link)`
  margin-bottom: ${props => props.theme.spacing.padding};
  font-size: ${props => props.theme.fontsize.small};
  font-weight: 400;
  text-decoration: none;
  ${breakpoint.desktop`
    flex: 0 0 ${props => (props.rowcount === 3 ? '33%' : '50%')};
  `};

  &:hover {
    color: ${props => props.theme.palette.primary};
  }
`;
