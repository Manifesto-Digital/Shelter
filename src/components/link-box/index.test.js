import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'gatsby';
import {
  snapshotComponent,
  mountWithTheme,
} from '../../../__tests__/helpers/index';
import LinkBox from './index';
import { ListItem } from './styles';
import { createFactory, createInternalLink } from '../../utils/test-factories';
import { hidePascalCaseWarning } from '../../utils/test-mocks';

export const createLinkBox = createFactory({
  headerText: 'What an amazing banner',
  itemsPerRow: 3,
  links: [
    createInternalLink(),
    createInternalLink(),
    createInternalLink(),
    createInternalLink(),
  ],
});

it('renders correctly', () => {
  const mockData = createLinkBox();

  snapshotComponent(<LinkBox data={mockData} />);
});

it('displays the correct header text', () => {
  const mockData = createLinkBox({
    headerText: 'Test header text',
  });
  const wrapper = shallow(<LinkBox data={mockData} />);
  expect(wrapper.find('h2').text()).toBe(mockData.headerText);
});

it('sets the number of links per row', () => {
  const mockData = createLinkBox({ itemsPerRow: 3 });
  const wrapper = shallow(<LinkBox data={mockData} />);
  expect(
    wrapper
      .find(ListItem)
      .at(0)
      .prop('rowCount')
  ).toBe(mockData.itemsPerRow);
});

hidePascalCaseWarning();

it('displays link and title correctly', () => {
  const mockData = createLinkBox({
    link: [
      {
        title: 'Shelter Demo Page',
        slug: 'shelter-demo-page',
      },
    ],
  });
  const wrapper = mountWithTheme(<LinkBox data={mockData} />);

  expect(
    wrapper
      .find(Link)
      .at(0)
      .text()
  ).toBe(mockData.links[0].title);

  expect(
    wrapper
      .find(Link)
      .at(0)
      .prop('to')
  ).toContain(mockData.links[0].slug);
});
