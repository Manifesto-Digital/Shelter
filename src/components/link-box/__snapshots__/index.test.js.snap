// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 40px 0;
  padding: 20px;
  background: #fff;
  box-shadow: 0 1.5px 4px rgba(0,0,0,0.12),0 1.5px 6px rgba(0,0,0,0.06);
  border-radius: 3px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c3 {
  margin-bottom: 20px;
  padding-right: 20px;
  font-size: 0.775rem;
  font-weight: 400;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  color: #ea232d;
}

@media (min-width:62em) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
}

@media (min-width:62em) {
  .c3 {
    -webkit-flex: 0 0 33.333333333333336%;
    -ms-flex: 0 0 33.333333333333336%;
    flex: 0 0 33.333333333333336%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h2>
      What an amazing banner
    </h2>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <Link
          to="/internal-link-slug"
        >
          Test Page Title
        </Link>
      </div>
      <div
        className="c3"
      >
        <Link
          to="/internal-link-slug"
        >
          Test Page Title
        </Link>
      </div>
      <div
        className="c3"
      >
        <Link
          to="/internal-link-slug"
        >
          Test Page Title
        </Link>
      </div>
      <div
        className="c3"
      >
        <Link
          to="/internal-link-slug"
        >
          Test Page Title
        </Link>
      </div>
    </div>
  </div>
</div>
`;
